addons/point_of_sale/models/pos_session.py

```diff
@@ -1851,7 +1851,10 @@ def _loader_params_res_users(self):
         }
 
     def _get_pos_ui_res_users(self, params):
-        user = self.env['res.users'].search_read(**params['search_params'])[0]
+        users = self.env['res.users'].search_read(**params['search_params'])
```

Comment:
I don't know what that method is supposed to do (didn't look), but simply in terms of implementation, it should probably be like this
```suggestion
        user = self.env['res.users'].search_read(**params['search_params'], limit=1)
```
I let you handle the rest with the pos team

Reply:
(You still need to `[0]` after the `if`, as `user` is supposed to be a `dict`, not a `list`)